clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -468,13 +530,37 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             String source = receive.source();
             InFlightRequest req = inFlightRequests.completeNext(source);
             AbstractResponse body = parseResponse(receive.payload(), req.header);
+            log.trace("Completed receive from node {}, for key {}, received {}", req.destination, req.header.apiKey(), body);
             if (req.isInternalMetadataRequest)
                 metadataUpdater.handleCompletedMetadataResponse(req.header, now, body);
-            else
+            else if (req.header.apiKey() == ApiKeys.API_VERSIONS.id) {
+                if (body instanceof ApiVersionsResponse)
```

Comment:
I think this check and the one above are redundant. Maybe we can remove the check for the api key?